### ISSUE TITLE
Missing counter data

### ISSUE THREAD

**Issue by heyman** [SamSaffron/graphite_docker#4]
Hi!

I've been using this image in production for a couple of weeks now, gathering various metrics on the health of my system. I'm sending data to StatsD over UDP, and I have a Grafana dashboard where I view the data. It's been working great with no issues until yesterday, when I suddenly started to get "gaps" in my counter metrics. 

Here's a couple of screenshots that shows graphs over a few different metrics for the same timespan:

![image](https://cloud.githubusercontent.com/assets/54217/7452621/497c7694-f260-11e4-9981-e254610990d7.png)

![image](https://cloud.githubusercontent.com/assets/54217/7452635/6e206afa-f260-11e4-8c8f-1b1202f7a6a4.png)

![image](https://cloud.githubusercontent.com/assets/54217/7452639/7b289a88-f260-11e4-9cf6-05c7920e252b.png)

![image](https://cloud.githubusercontent.com/assets/54217/7452929/ea251be8-f263-11e4-8398-bc6119e077c6.png)

In the first two graphs - whose metric type are counters - there are two large gaps. However, the third and fourth graph - whose metrics are timer and gauge - does *not* have any gaps.

Does anyone have any idea of what might be going on?

What I've checked so far:

* Disk IO is low (An answer to http://serverfault.com/questions/533198/graphite-stops-collecting-data-randomly seems to suggest that one could see similar symptoms if an IOPS bottleneck is encountered)
* Since it's still collecting timer and gauge data, it seems unlikely to be a network problem.
* The following script reported no whisper file corruptions: https://gist.github.com/gonsfx/4111791

**Comment by SamSaffron** [comment]
I think gauge and timers fill up gaps which, this very much looks like comms are going down somehow between the container and reporter ... maybe try running a simulation script locally to see if it works during these outages?

**Comment by heyman** [comment]
Whenever I do a deploy I increment a counter, and I've set up Grafana to display annotations in the graphs according to this counter. These deploy annotations gets properly reported in the graphs during the outages (see the blue veritcal line in screenshot 3 & 4 below), so it seems unlikely to be a communications error between the reporter and the container.

However, I've noticed another weird thing, which is that during the outages, I can see data for the last 10 second period (It's only noticeable if I select a time range that makes the X-axis steps small enough). Here are two screenshots, taken ~2 minutes after each other, that shows this (the red vertical line is at the same point in time in both graphs).

![image](https://cloud.githubusercontent.com/assets/54217/7473823/0de8123a-f33e-11e4-9119-1ad0c7a954d2.png)

![image](https://cloud.githubusercontent.com/assets/54217/7473835/2bb9d51e-f33e-11e4-974a-4e30bd13e52e.png)

And here are two screenshots that shows the same thing for the `stats.statsd.metrics_received` graph. 

![image](https://cloud.githubusercontent.com/assets/54217/7473874/677d78a8-f33e-11e4-8954-7e7b59bc33d7.png)

![image](https://cloud.githubusercontent.com/assets/54217/7473898/86f51ba0-f33e-11e4-81cd-8fcdc6163e96.png)

So it seems that somehow the data for the last reported time period is retrievable, but the data isn't persisted once another report comes in?

**Comment by heyman** [comment]
Hi again!

I believe I might have found what was causing the issue. I found stray processes from an old container that hadn't been properly killed (due to the following docker issue: https://github.com/docker/docker/issues/12738).

My current theory is that one of the old statsd/graphite/carbon processes were sometimes overwriting the last reported metrics with 0 values.

I've just recently killed the stray processes, and so far it looks good. So with the risk of counting my chickens before they hatch, I'm closing this issue. Sorry for wasting your time with a non-related issue. Hopefully it can help someone else who might find this page through Google.
